src/main/java/org/opensearch/knn/common/featureflags/KNNFeatureFlags.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ */
+
+package org.opensearch.knn.common.featureflags;
+
+import com.google.common.annotations.VisibleForTesting;
+import lombok.experimental.UtilityClass;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.knn.index.KNNSettings;
+
+import java.util.stream.Stream;
+
+import static org.opensearch.common.settings.Setting.Property.Dynamic;
+import static org.opensearch.common.settings.Setting.Property.NodeScope;
+
+/**
+ * Class to manage KNN feature flags
+ */
+@UtilityClass
+public class KNNFeatureFlags {
+
+    // Feature flags
+    private static final String KNN_LAUNCH_QUERY_REWRITE_ENABLED = "knn.feature.query.rewrite.enabled";
+    private static final boolean KNN_LAUNCH_QUERY_REWRITE_ENABLED_DEFAULT = true;
+
+    @VisibleForTesting
+    public static final Setting<Boolean> KNN_LAUNCH_QUERY_REWRITE_ENABLED_SETTING = Setting.boolSetting(
+        KNN_LAUNCH_QUERY_REWRITE_ENABLED,
+        KNN_LAUNCH_QUERY_REWRITE_ENABLED_DEFAULT,
+        NodeScope,
+        Dynamic
+    );
+
+    public static Stream<Setting<?>> getFeatureFlags() {
+        return Stream.of(KNN_LAUNCH_QUERY_REWRITE_ENABLED_SETTING);
+    }
+
+    public static boolean isFeatureEnabled(final String featureName) {
```

Comment:
Are we planning to use this function in future for any other feature? Right now with flow it seems redundant , since another public function call another public? and in future if need this to avoid feature name to function ,it could be private.

Reply:
> Are we planning to use this function in future for any other feature?

Its made for reuse so yes 

> Right now with flow it seems redundant , since another public function call another public?

if anything I think `isKnnLaunchQueryRewriteEnabled` is redundant IMO.  With each feature we add there will be one extra method and can turn into a long list of methods. There is nothing wrong in forcing the user to know what setting needs to be used and its also hardly reusable but it's not a hill to die on so I have added it for now. 

I would rather prefer reusability and maintainability over convenience and get rid of `isKnnLaunchQueryRewriteEnabled` method than making this private.

Reply:
yeah that was my concern , either of two has to be here , not both , if you are removing isKnnLaunchQueryRewriteEnabled , that's also fine by me